Sources/mpc-core-kit-swift/mpcCoreKitSwift.swift

```diff
@@ -140,12 +142,43 @@ public struct MpcCoreKit  {
         return try await self.login(userData: TorusKeyData(torusKey: torusKey, userInfo: modUserInfo))
     }
     
+    public func getUserInfo() throws -> [String: Any] {
+        guard let userInfo = self.userInfo else {
+            throw ("user is not logged in.")
+        }
+        return userInfo
+    }
+    
+    public func getKeyDetails() async throws -> MpcKeyDetails {
+        guard let finalKeyDetails = try self.tkey?.get_key_details() else {
```

Comment:
Not sure that we want to mix details of the metadata key with the signing key, these are different things?

Consider separate methods. Also, reconstructing the key would return KeyDetails by default, so why the need to fetch it again?

Reply:
in mpc web we have the same logic to get tss pub and attach
for ref
`  public getKeyDetails(): MPCKeyDetails {
    this.checkReady();
    const tkeyDetails = this.tKey.getKeyDetails();
    const tssPubKey = this.state.tssPubKey ? Point.fromBufferSEC1(this.state.tssPubKey).toTkeyPoint() : undefined;

    const factors = this.tKey.metadata.factorPubs ? this.tKey.metadata.factorPubs[this.tKey.tssTag] : [];
    const keyDetails: MPCKeyDetails = {
      // use tkey's for now
      requiredFactors: tkeyDetails.requiredShares,
      threshold: tkeyDetails.threshold,
      totalFactors: factors.length + 1,
      shareDescriptions: this.tKey.getMetadata().getShareDescription(),
      metadataPubKey: tkeyDetails.pubKey,
      tssPubKey,
    };
    return keyDetails;
  }`

Reply:
Yes I get that. It doesn't answer the original questions though. This may also need to be changed there depending on the answers here.

Reply:
Let get it match with the web version first. (v2)
we can file pr and discuss if the changes is needed for mpc core kit web ( especially v3 that is in working)

Reply:
@ieow Ok, approve it if you're happy with it. Think we'll have to come back it change it though, however not blocking this PR from going in.